### PR TITLE
MusicDataBaseDirectory::NODE_TYPE: convert to enum class

### PR DESCRIPTION
**Pull request by notspiff**
## Description
Rename and change an enum to an enum class

## Motivation and context
More modern, more type safe, ..

## How has this been tested?
It builds

## What is the effect on users?
None

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Review by ksooo — COMMENTED** [review]
DirectoryNode.h is included in many header files. Any chance to replace some of the includes with forward declarations to the new enum class now?

Quick check:
* filesystem/MusicdatabaseDirectory/QueryParams.h
* filesystem/VideodatabaseDirectory/QueryParams.h
* filesystem/MusicDatabaseDirectory.h
* filesystem/VideoDatabaseDirectory.h

File: xbmc/filesystem/MusicDatabaseDirectory.cpp [hunk 1]
```diff
@@ -271,14 +270,21 @@ bool CMusicDatabaseDirectory::GetLabel(const std::string& strDirectory, std::str
...
 {
-  MUSICDATABASEDIRECTORY::NODE_TYPE type = GetDirectoryChildType(path);
-  if (type == MUSICDATABASEDIRECTORY::NODE_TYPE_SONG) return true;
-  if (type == MUSICDATABASEDIRECTORY::NODE_TYPE_SINGLES) return true;
-  if (type == MUSICDATABASEDIRECTORY::NODE_TYPE_ALBUM_RECENTLY_ADDED_SONGS) return true;
-  if (type == MUSICDATABASEDIRECTORY::NODE_TYPE_ALBUM_RECENTLY_PLAYED_SONGS) return true;
-  if (type == MUSICDATABASEDIRECTORY::NODE_TYPE_ALBUM_TOP100_SONGS) return true;
-  if (type == MUSICDATABASEDIRECTORY::NODE_TYPE_SONG_TOP100) return true;
-  if (type == MUSICDATABASEDIRECTORY::NODE_TYPE_DISC) return true;
+  auto type = GetDirectoryChildType(path);
```
[the remark this answers is not in the document]
Comment (notspiff) [reply xbmc/filesystem/MusicDatabaseDirectory.cpp]: added the const.

**Review by garbear — APPROVED** [review]
LGTM

**Review by ksooo — APPROVED** [review]
Shove it in.

**Pull request merged**